### PR TITLE
Adds promote workflow files to the default branch

### PR DESCRIPTION
**Pull request by saltiyazan**
# Description

wokflow_dispatch workflow files are only visible under the actions tab if they are on the default branch. This PR brings back promote workflow files to the main branch too.

Note: the files should also be available on the release branches for the action to run correctly.

## Checklist

- [ ] My code follows the [style guidelines](/CONTRIBUTING.md) of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added tests that validate the behaviour of the software.
- [ ] I validated that new and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I have bumped the version of any required library.

### CODE REVIEW

**Review by ghislainbourgeois — APPROVED** [review]

File: .github/workflows/bundle.promote.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        run: pip install -U Jinja2
+      - name: render bundle.yaml
+        run: |
+          python orc8r-bundle/render_bundle.py \
+          --template orc8r-bundle/bundle.yaml.j2 --output orc8r-bundle/bundle.yaml --channel ${{ env.promote-to }}
+      - name: copy README.md
+        run: |
+          cp README.md orc8r-bundle/
+      - name: Upload bundle
+        uses: canonical/charming-actions/upload-bundle@1.0.2
```
Comment (sanchezfdezjavier) [inline .github/workflows/bundle.promote.yaml]: We need to update the version of the charming-action.

File: .github/workflows/charms.promote.yaml [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+          elif [ "${PROMOTION}" == "stable -> latest/stable" ]; then
+            echo "promote-from=${{ github.ref_name }}/stable" >> ${GITHUB_ENV}
+            echo "promote-to=latest/stable" >> ${GITHUB_ENV}
+          fi
+      - name: Set base channel
+        if: matrix.charms == 'nms-nginx-proxy'
+        run: echo "BASE_CHANNEL=22.04" >> ${GITHUB_ENV}
+
+      - name: Promote all Charms
+        uses: canonical/charming-actions/release-charm@2.2.2
```
Comment (sanchezfdezjavier) [inline .github/workflows/charms.promote.yaml]: We need to update the version of the charming-action.

File: .github/workflows/charms.promote.yaml [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+          elif [ "${PROMOTION}" == "stable -> latest/stable" ]; then
+            echo "promote-from=${{ github.ref_name }}/stable" >> ${GITHUB_ENV}
+            echo "promote-to=latest/stable" >> ${GITHUB_ENV}
+          fi
+      - name: Set base channel
+        if: github.event.inputs.charm-name == 'nms-nginx-proxy'
+        run: echo "BASE_CHANNEL=22.04" >> ${GITHUB_ENV}
+
+      - name: Promote Charm
+        uses: canonical/charming-actions/release-charm@2.2.2
```
Comment (sanchezfdezjavier) [inline .github/workflows/charms.promote.yaml]: We need to update the version of the charming-action.

File: .github/workflows/bundle.promote.yaml (hunk 1 above)
Comment (saltiyazan) [reply .github/workflows/bundle.promote.yaml]: ✅

File: .github/workflows/charms.promote.yaml (hunk 2 above)
Comment (saltiyazan) [reply .github/workflows/charms.promote.yaml]: ✅

File: .github/workflows/charms.promote.yaml (hunk 3 above)
Comment (saltiyazan) [reply .github/workflows/charms.promote.yaml]: ✅

**Review by Gmerold — APPROVED** [review]

**Review by sanchezfdezjavier — APPROVED** [review]
LGTM

**Pull request merged**